pkg/kubectl/cmd/util/printing.go

```diff
@@ -126,6 +127,15 @@ func PrinterForCommand(cmd *cobra.Command, outputOpts *printers.OutputOptions, m
 		return nil, err
 	}
 
+	// if printer.isGeneric() not true, it must be a HumanReadablePrinter
+	// we execute AddHandlers() here before maybeWrapSortingPrinter so that we don't
+	// need to use delegatePrinter convert again in ring2Factory.PrinterForMapping when it's a SortingPrinter
+	if !printer.IsGeneric() {
+		if humanReadablePrinter, ok := printer.(*printers.HumanReadablePrinter); ok {
```

Comment:
To continue our previous conversation: I would write this:

```go
if humanReadablePrinter, ok := printer.(*printers.HumanReadablePrinter); ok {
    printersinternal.AddHandlers(humanReadablePrinter)
}
```

Why do you need the IsGeneric check? If the printer is HumanReadable, you know the printer is nongeneric.